src/dist/mod.rs

```diff
@@ -97,6 +97,20 @@ Then you can use the toolchain with commands such as:
 
     cargo +nightly-2018-12-27 build"
         );
+    } else if ["beta", "stable"].iter().any(|&p| toolchain.starts_with(p)) {
+        let _ = write!(
+            buf,
+            "\
+One or many components listed above might have been permanently removed from newer versions
```

Comment:
The wording here seems a little vague. Can we do something more specific, perhaps by checking for the specific `rls` component?

Reply:
@djc I intended to learn from the message right above and to add a link pointing to a list of removed components somewhere online (see the `Concerns` section of the original post). As a result, not mentioning `rls` explicitly here is intentional, since this gives the release team the freedom to remove components in the middle of any Rustup release cycle once this fix lands, as long as the online list can be updated in time.

Reply:
@djc I've decided to use https://rust-lang.github.io/rustup/devel/concepts/components.html?highlight=remove#previous-components, and I've updated this message to include that link.